tests/sentry/api/endpoints/test_organization_codeowners.py

```diff
@@ -0,0 +1,147 @@
+        for code_owner in response.data:
+            # Check error object shape
+            assert "codeMappingId" in code_owner.keys()
+            assert "errors" in code_owner.keys()
+            for field in [
+                "missing_external_users",
+                "missing_external_teams",
+                "users_without_access",
+                "teams_without_access",
+            ]:
+                assert field in code_owner["errors"].keys()
+            # Check individual code owners for their errors
+            if int(code_owner["codeMappingId"]) == self.data_1["codeMappingId"]:
+                assert "@hernando" in code_owner["errors"]["missing_external_users"]
+
+            if int(code_owner["codeMappingId"]) == self.data_2["codeMappingId"]:
+                assert "@hernando" in code_owner["errors"]["missing_external_users"]
+                assert "@sean" in code_owner["errors"]["missing_external_users"]
+                assert "#executives" in code_owner["errors"]["teams_without_access"]
+
+    def test_no_access_to_project(self):
+        """
+        Tests that projects the requesting user does not have access to are not in the response
+        """
+        self.create_codeowners(self.project_1, self.code_mapping_1, raw=self.data_1["raw"])
+        self.create_codeowners(self.project_2, self.code_mapping_2, raw=self.data_2["raw"])
+        # Create a project/code owners that user_1 doesn't have access to
+        project_3 = self.create_project(
+            organization=self.organization, teams=[self.team_2], slug="fire-everyone"
+        )
+        code_mapping_3 = self.create_code_mapping(project=project_3)
+        code_owners = self.create_codeowners(project_3, code_mapping_3, raw="all/*  @life/exec\n")
+        response = self.get_success_response(self.organization.slug, status=status.HTTP_200_OK)
+        # Check the original two code owners are still visible
+        assert len(response.data) == 2
+        assert (
+            serialize(
+                code_owners,
+                self.user,
+                serializer=ProjectCodeOwnersSerializer(expand=["errors"]),
+            )
+            not in response.data
+        )
+        # Check that a user who can see all the projects receives them all
+        self.login_as(self.user_2)
+        response = self.get_success_response(self.organization.slug, status=status.HTTP_200_OK)
+        assert len(response.data) == 3
+        assert (
+            serialize(
+                code_owners,
+                self.user,
+                serializer=ProjectCodeOwnersSerializer(expand=["errors"]),
+            )
+            in response.data
+        )
+        # Check that a user with higher perms (but not explicitly on the teams) can see all as well
+        self.create_member
+        self.login_as(self.owner)
+        response = self.get_success_response(self.organization.slug, status=status.HTTP_200_OK)
+        assert len(response.data) == 3
```

Comment:
This part of the test is failing until I can figure out my TODO in the PR description.